eclair-core/src/main/scala/fr/acinq/eclair/payment/send/PaymentLifecycle.scala

```diff
@@ -390,8 +393,11 @@ object PaymentLifecycle {
   sealed trait SendPayment {
     // @formatter:off
     def replyTo: ActorRef
-    def finalPayload: FinalPayload
-    def extraEdges: Seq[ExtraEdge]
+    /** Amount to send to the target node. */
+    def amount: MilliSatoshi
+    /** Final payment recipient. */
+    def recipient: Recipient
+    /** Target node (may be the recipient when using source-routing, or the first trampoline node when using trampoline). */
```

Comment:
Because we need to ask the `Router` to find a route to a given `nodeId`, and in the case of trampoline it has to be the first trampoline. I don't understand how you would do this differently, can you detail?

Note that I plan on removing that in the future, it's a clear sign that something isn't done properly. In a future PR, I will add a `TrampolineEdge` similar to `BlindedEdge`, which will let us inject it into the graph and simplify the payment FSMs (and trivially add support for blinded trampoline), but that deserves a separate PR. I mention this in the description of #2481 since it will use a similar mechanism.

Reply:
I've moved that to the trampoline recipient itself, since we cannot use just inject "trampoline edges" into the graph because it doesn't work well with MPP aggregation at trampoline nodes (fees would be counted multiple times, as is explained in the comment in the code).